pinot-core/src/main/java/org/apache/pinot/core/realtime/converter/RealtimeSegmentConverter.java

```diff
@@ -54,10 +54,12 @@
   private List<String> invertedIndexColumns;
   private List<String> noDictionaryColumns;
   private StarTreeIndexSpec starTreeIndexSpec;
+  private List<String> varLengthDictionaryColumns;
 
   public RealtimeSegmentConverter(MutableSegmentImpl realtimeSegment, String outputPath, Schema schema,
       String tableName, String timeColumnName, String segmentName, String sortedColumn,
-      List<String> invertedIndexColumns, List<String> noDictionaryColumns, StarTreeIndexSpec starTreeIndexSpec) {
+      List<String> invertedIndexColumns, List<String> noDictionaryColumns,
+      List<String> varLengthDictionaryColumns, StarTreeIndexSpec starTreeIndexSpec) {
```

Comment:
I am not clear on why RT code needs to know about varLengthDictColumns, if we are not supporting them in this PR.

Reply:
We need to know this information while converting realtime segments into offline segments. Do you have any suggestions to get that without touching these realtime classes?

Reply:
I guess if RealtimeSegmentConverter took a segmentGeneratorConfig as param, then this would have been cleaner. May be file an issue for cleanup, if you agree?

Reply:
Instead, we should deprecate segmentGenerationConfig and use TableConfig everywhere.

Reply:
I've not spent enough time on this part of code to give my take on what's the best thing to do.

If one of you can please log an issue with more details, I'll take it up next, check the code and can make the changes finally.

Reply:
This is outside the scope of this PR. Dont worry about it.